test-framework/junit5/src/main/java/io/quarkus/test/junit/AppMakerHelper.java

```diff
@@ -0,0 +1,352 @@
+import io.quarkus.bootstrap.app.StartupAction;
+import io.quarkus.bootstrap.model.ApplicationModel;
+import io.quarkus.bootstrap.resolver.AppModelResolverException;
+import io.quarkus.bootstrap.runner.Timing;
+import io.quarkus.bootstrap.utils.BuildToolHelper;
+import io.quarkus.bootstrap.workspace.ArtifactSources;
+import io.quarkus.bootstrap.workspace.SourceDir;
+import io.quarkus.bootstrap.workspace.WorkspaceModule;
+import io.quarkus.commons.classloading.ClassLoaderHelper;
+import io.quarkus.paths.PathList;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.test.common.PathTestHelper;
+import io.quarkus.test.common.RestorableSystemProperties;
+import io.quarkus.test.common.TestClassIndexer;
+
+public class AppMakerHelper {
+
+    // TODO Copied from superclass of thing we copied
+    protected static final String TEST_LOCATION = "test-location";
+    protected static final String TEST_CLASS = "test-class";
+    protected static final String TEST_PROFILE = "test-profile";
+    /// end copied
+
+    private static Class<?> quarkusTestMethodContextClass;
+    private static boolean hasPerTestResources;
+
+    private static List<Object> testMethodInvokers;
+    private Runnable configCleanup;
+
+    public static class PrepareResult {
+        protected final AugmentAction augmentAction;
+        public final QuarkusTestProfile profileInstance;
+        protected final CuratedApplication curatedApplication;
+        public final Path testClassLocation;
+
+        public PrepareResult(AugmentAction augmentAction, QuarkusTestProfile profileInstance,
+                CuratedApplication curatedApplication, Path testClassLocation) {
+            System.out.println("PrepareResult" + augmentAction + ": " + profileInstance + " test class " + testClassLocation);
+
+            this.augmentAction = augmentAction;
+            this.profileInstance = profileInstance;
+            this.curatedApplication = curatedApplication;
+            this.testClassLocation = testClassLocation;
+        }
+    }
+
+    public static ApplicationModel getGradleAppModelForIDE(Path projectRoot) throws IOException, AppModelResolverException {
+        return System.getProperty(BootstrapConstants.SERIALIZED_TEST_APP_MODEL) == null
+                ? BuildToolHelper.enableGradleAppModelForTest(projectRoot)
+                : null;
+    }
+
+    private PrepareResult createAugmentor(final Class<?> requiredTestClass, String displayName, boolean isContinuousTesting,
+            CuratedApplication curatedApplication,
+            Class<? extends QuarkusTestProfile> profile,
+            Collection<Runnable> shutdownTasks) throws Exception {
+
+        if (curatedApplication == null) {
+            curatedApplication = makeCuratedApplication(requiredTestClass, displayName, isContinuousTesting, shutdownTasks);
+        }
+        Path testClassLocation = getTestClassLocationIncludingPossibilityOfGradleModel(requiredTestClass);
```

Comment:
It seems like `testClassLocation` should already be available from the `curatedApplication` here? Could they be different here? If they are, isn't it an issue?

Reply:
You're totally right! Let me try that out.

Reply:
Implemented in https://github.com/quarkusio/quarkus/pull/34681/commits/9dcd37826e182b78402c4119c1399e66a407de67, although it turns out to only be a marginal code saving. Testing in https://github.com/quarkusio/quarkus/actions/runs/13845822559